libs/prelude/Prelude/Uninhabited.idr

```diff
@@ -34,8 +34,8 @@ absurd : Uninhabited t => (h : t) -> a
 absurd h = void (uninhabited h)
 
 public export
-Uninhabited (True = False) where
-  uninhabited Refl impossible
+Uninhabited (x = y) => Uninhabited (y = x) where
```

Comment:
This seems like a great way to make the proof search loop.

Reply:
```idris
module Test

Uninhabited (x = y) => Uninhabited (y = x) where
  uninhabited = \h => void $ uninhabited $ sym h

data MyNat = ZZ | SS MyNat

-- Comment out for attempting to trap the compiler into an infinity loop
-- Uninhabited (ZZ = SS ZZ) where
--   uninhabited Refl impossible 

nonSense : Not (SS ZZ = ZZ)
nonSense = absurd
```

outpus:

```
Error: While processing right hand side
of nonSense. Can't find an implementation
for Uninhabited (SS ZZ = ZZ).

Test:12:12--12:18
 08 | -- Uninhabited (ZZ = SS ZZ) where
 09 | --   uninhabited Refl impossible 
 10 | 
 11 | nonSense : Not (SS ZZ = ZZ)
 12 | nonSense = absurd
                 ^^^^^^

Error(s) building file Test.idr
```

Reply:
Seems that the complier is robust enough to not get trapped.

Reply:
well... the test shows that the proof search indeed looped.